sensorwebclient-sos-rest/src/main/java/org/n52/series/api/proxy/v1/srv/RawObservationDataService.java

```diff
@@ -0,0 +1,413 @@
+import org.slf4j.LoggerFactory;
+
+import net.opengis.gml.x32.FeatureCollectionDocument;
+import net.opengis.gml.x32.FeatureCollectionType;
+import net.opengis.om.x20.OMObservationPropertyType;
+import net.opengis.sos.x20.GetObservationResponseDocument;
+import net.opengis.sos.x20.GetObservationResponseType.ObservationData;
+
+/**
+ * Process raw data query for observations/timeseries.
+ * 
+ * @author Carsten Hollmann <c.hollmann@52north.org>
+ * @since 1.7.3
+ *
+ */
+public abstract class RawObservationDataService implements RawDataService {
+
+    static final Logger LOGGER = LoggerFactory.getLogger(RawObservationDataService.class);
+
+    private SimpleDateFormat dateFormat = createIso8601Formatter();
+
+    private static XmlOptions xmlOptions = initXmlOptions();
+
+    private static XmlOptions initXmlOptions() {
+        XmlOptions xmlOptions = new XmlOptions();
+        Map<String, String> prefixMap = new HashMap<String, String>();
+        prefixMap.put("http://www.opengis.net/gml/3.2", "gml");
+        prefixMap.put("http://www.opengis.net/om/2.0", "om");
+        prefixMap.put("http://www.opengis.net/waterml/2.0", "wml2");
+        prefixMap.put("http://www.opengis.net/waterml-dr/2.0", "wml2dr");
+        prefixMap.put("http://www.opengis.net/sampling/2.0", "sf");
+        prefixMap.put("http://www.opengis.net/samplingSpatial/2.0", "sams");
+        prefixMap.put("http://www.w3.org/1999/xlink", "xlink");
+        prefixMap.put("http://www.w3.org/2001/XMLSchema-instance", "xsi");
+        prefixMap.put("http://www.w3.org/2001/XMLSchema", "xs");
+        xmlOptions.setSaveSuggestedPrefixes(prefixMap);
+        xmlOptions.setSaveImplicitNamespaces(prefixMap);
+        xmlOptions.setSaveAggressiveNamespaces();
+        xmlOptions.setSavePrettyPrint();
+        xmlOptions.setSaveNamespacesFirst();
+        xmlOptions.setCharacterEncoding("UTF-8");
+        return xmlOptions;
+    }
+
+    @Override
+    public InputStream getRawData(String id, IoParameters query) {
+        return getRawData(createForSingleTimeseries(id, query));
+    }
+
+    @Override
+    public InputStream getRawData(UndesignedParameterSet parameters) {
+        checkRawDataFormat(parameters.getRawFormat(), parameters.getTimeseries());
+        Map<SOSMetadata, Set<String>> timeseriesMetadataMap = getTimeseriesMetadataMap(parameters);
+        if (timeseriesMetadataMap.isEmpty()) {
+            throw new ResourceNotFoundException("Found no data for timeseries and parameter.");
+        } else if (timeseriesMetadataMap.size() > 1) {
+            throw new BadRequestException("Querying raw timeseries data from several services is not yet supported");
+        }
+        List<OperationResult> result = queryObservationsForRequestedParameter(timeseriesMetadataMap, parameters);
+        if (result.isEmpty()) {
+            LOGGER.error("Get no result for GetObservation request");
+            return null;
+        }
+        // TODO implement support for multiple results
```

Comment:
do you have an idea how to support raw observation data of multiple timeseries at all without kind of merging the responses? I suspect that even multipart messages [wouldn't give a sufficient solution](http://stackoverflow.com/a/1041591/2299448) to that issue.